tests/generate/test_generate_custom_generate_commands.py

```diff
@@ -0,0 +1,125 @@
+from collections.abc import Callable
+from pathlib import Path
+
+from _pytest.tmpdir import TempPathFactory
+from algokit.core.conf import ALGOKIT_CONFIG
+from pytest_mock import MockerFixture
+
+from tests.utils.approvals import verify
+from tests.utils.click_invoker import invoke
+
+DirWithAppSpecFactory = Callable[[Path], Path]
+
+
+def test_generate_custom_generate_commands_no_toml(tmp_path_factory: TempPathFactory) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+
+    result = invoke("generate", cwd=cwd)
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+def test_generate_custom_generate_commands_invalid_generic_generator(tmp_path_factory: TempPathFactory) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+
+    (cwd / ALGOKIT_CONFIG).write_text(
+        """
+[generate]
+description = "invalid"
+path = "invalid"
+    """.strip(),
+        encoding="utf-8",
+    )
+
+    result = invoke("generate", cwd=cwd)
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+def test_generate_custom_generate_commands_valid_generator(
+    tmp_path_factory: TempPathFactory,
+) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+    (cwd / "smart_contract").mkdir()
+    # Required for windows compatibility
+    smart_contract_path = str((cwd / "smart_contract").absolute()).replace("\\", r"\\")
```

Comment:
the above 3 lines are copy pasted a few times, maybe have a fixture and/or helper function? we don't need to overthink / over engineer the tests, but if a single helper function aids readability and understanding then we should do that.

Reply:
@achidlow done - added a simple fixture to encapsulate these 3 lines and return cwd, custom_folder_path as a tuple👍